guides/source/routing.md

```diff
@@ -18,51 +18,66 @@ After reading this guide, you will know:
 The Purpose of the Rails Router
 -------------------------------
 
-The Rails router recognizes URLs and dispatches them to a controller's action, or to a Rack application. It can also generate paths and URLs, avoiding the need to hardcode strings in your views.
+The Rails router matches incoming HTTP requests to specific controller actions
+in your Rails application based on the URL path. (It can also forward to a
+[Rack](rails_on_rack.html) application.) The router also generates path and URL
+helpers based on the resources configured in the router.
 
-### Connecting URLs to Code
+### Routing Incoming URLs to Code
 
-When your Rails application receives an incoming request for:
+When your Rails application receives an incoming request, it asks the router to match it to a controller action. For example, take the following incoming request:
 
 ```
-GET /patients/17
+GET /users/17
 ```
 
-it asks the router to match it to a controller action. If the first matching route is:
+The Router's job is to match it to a controller action. If the first matching route is:
 
 ```ruby
-get '/patients/:id', to: 'patients#show'
+get '/users/:id', to: 'user#show'
 ```
 
-the request is dispatched to the `patients` controller's `show` action with `{ id: '17' }` in `params`.
+The request is matched to the `UsersController` class's `show` action with `{ id: '17' }` in the `params` hash.
+
+The `to:` option expects a `controller#action` format when passed a string. Alternatively, You can pass a symbol and use the `action:` option, instead of `to:`. You can also pass a string without a `#`, in which case the `controller:` option is used instead to `to:`. For example:
+
+```ruby
+get '/users/:id', action: :show, controller: 'users'
```

Comment:
How about passing the controller first, then the action? I think keeping the same order as `controller#action` helps people to visualize what changed.
```suggestion
get '/users/:id', controller: 'users', action: :show
```